torchaudio/datasets/legacy/vctk.py

```diff
@@ -0,0 +1,243 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import torch.utils.data as data
+import os
+import os.path
+import shutil
+import errno
+import torch
+import torchaudio
+
+AUDIO_EXTENSIONS = [
+    '.wav', '.mp3', '.flac', '.sph', '.ogg', '.opus',
+    '.WAV', '.MP3', '.FLAC', '.SPH', '.OGG', '.OPUS',
+]
+
+
+def is_audio_file(filename):
+    return any(filename.endswith(extension) for extension in AUDIO_EXTENSIONS)
+
+
+def make_manifest(dir):
+    audios = []
+    dir = os.path.expanduser(dir)
+    for target in sorted(os.listdir(dir)):
+        d = os.path.join(dir, target)
+        if not os.path.isdir(d):
+            continue
+
+        for root, _, fnames in sorted(os.walk(d)):
+            for fname in fnames:
+                if is_audio_file(fname):
+                    path = os.path.join(root, fname)
+                    item = path
+                    audios.append(item)
+    return audios
+
+
+def read_audio(fp, downsample=True):
+    if downsample:
+        E = torchaudio.sox_effects.SoxEffectsChain()
+        E.set_input_file(fp)
+        E.append_effect_to_chain("gain", ["-h"])
+        E.append_effect_to_chain("channels", [1])
+        E.append_effect_to_chain("rate", [16000])
+        E.append_effect_to_chain("gain", ["-rh"])
+        E.append_effect_to_chain("dither", ["-s"])
+        sig, sr = E.sox_build_flow_effects()
+    else:
+        sig, sr = torchaudio.load(fp)
+    sig = sig.contiguous()
+    return sig, sr
+
+
+def load_txts(dir):
```

Comment:
I think you could first run a "find" function to get a dictionary of file paths plus the corresponding text (in UTF-8) format.

Then somewhere you need a map between transcription files and audio files.

Reply:
This is part of the legacy datasets which I simply moved in a legacy folder.